cpu/stm32_common/periph/dac.c

```diff
@@ -57,12 +88,8 @@ int8_t dac_init(dac_t line)
 void dac_set(dac_t line, uint16_t value)
 {
     value = (value >> 4);       /* scale to 12-bit */
-    if (dac_config[line].chan) {
-        DAC->DHR12R2 = value;
-    }
-    else {
-        DAC->DHR12R1 = value;
-    }
+
+    DAC_WRITE_OUTPUT(line, value);
```

Comment:
hm, just checked, seems we have 3 different configurations:
- 1 DAC, 1 channel
- 1 DAC, 2 channels
- 2 DACs, 2 channels each

```
RIOT/cpu $ git grep RCC_APB1ENR_DAC
stm32f0/include/stm32f051x8.h:2070:#define  RCC_APB1ENR_DACEN
stm32f0/include/stm32f091xc.h:3865:#define  RCC_APB1ENR_DACEN
stm32f1/include/stm32f103xe.h:1399:#define  RCC_APB1ENR_DACEN
stm32f3/include/stm32f303xc.h:4668:#define  RCC_APB1ENR_DAC1EN
stm32f3/include/stm32f303xe.h:5740:#define  RCC_APB1ENR_DAC1EN
stm32f3/include/stm32f334x8.h:5557:#define  RCC_APB1ENR_DAC2EN
stm32f3/include/stm32f334x8.h:5559:#define  RCC_APB1ENR_DAC1EN
stm32f4/include/stm32f407xx.h:4991:#define  RCC_APB1ENR_DACEN  
stm32f4/include/stm32f415xx.h:5005:#define  RCC_APB1ENR_DACEN  
stm32l1/include/stm32l1xx.h:3722:#define  RCC_APB1ENR_DACEN      
```

so maybe it is not as easy as I thought... I think the solution is probably to put some kind of define in the specific `stm32xx/include/periph_cpu.h` files and use it here to differentiate between those options listed above.